### PR TITLE
[#2172] Extend API docs

### PR DESCRIPTION
**Pull request by jswk**
Add more detailed API docs.

Closes: #2172.

### CODE REVIEW

**Comment by abacz** [comment]
@jarekzet ive prepared few previews for this PR
- https://projects.invisionapp.com/share/JXOZKC8WMDU#/441030132_MyProfile_-_API_-_Token_Generated
- https://projects.invisionapp.com/share/JXOZKC8WMDU#/441030133_MyProfile_-_API_-_No_Token
- https://projects.invisionapp.com/share/JXOZKC8WMDU#/454770927_MyProfile_-_API_-_Token_Generated_Hidden

**Comment by Marcelinna** [comment]
image

maybe you could write something about this model, project_item often appears in the API description and in my opinion there is no description of it. There is a diagram but maybe there could be more to write about it. What do you think about it?maybe it should be selected that this is the data model for the Ordering API?

**Comment by Marcelinna** [comment]
image

this applies only to the ordering Api, maybe it should be emphasized, somehow reading the descriptions, sometimes I miss information about which API this applies to.
project_item appears here many times, for a person who does not know the application very well, it can be a bit unclear. ?
I am missing an introduction to the default OMS, primary OMS, secondary OMS.

File: app/views/api_docs/subsections/_basic_information.html.haml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  \.
+
+%p
+  The filter is exclusive (i.e. for
+  %code from_id=1
+  one gets ids 2, 3, etc.).
+
+%h3 Trigger
+%p
+  For an OMS adapter to receive asynchronous notifications about new and changed resources, it has to expose a trigger
```
Comment (bwilk) [inline app/views/api_docs/subsections/_basic_information.html.haml:103]: In order to receive asynchronous notifications about new and changed resources, an OMS needs to expose a  HTTP trigger endpoint. HTTP GET??? request will be issued on this endpoint each time the OMS needs to be notified of the changes.

**Comment by jswk** [comment]
I added a little more info about the data model: https://github.com/cyfronet-fid/marketplace/pull/2188#issuecomment-882302690.
I'll go through the entire text, including the [problems in access model](https://github.com/cyfronet-fid/marketplace/pull/2188#issuecomment-882308360).

**Comment by Marcelinna** [comment]
image
 link for Ordering API redirect  takes to the page with Offering API

File: app/views/api_docs/subsections/_basic_information.html.haml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  %code GET /events
+  endpoint for updates.
+
+%h4 Data model
+%p
+  In the EOSC Marketplace every user can create Projects to organize their actions and get guidance from the operations
+  team.
+%p
+  Within a Project, the user will create resource orders and open-access resources. Both are represented as a single
+  resource type in the API: the Project Item.
```
Comment (Marcelinna) [inline app/views/api_docs/subsections/_basic_information.html.haml:158]: create resource orders and open-access resources ?
do you think about adding resource to your project? orders?

Comment (Marcelinna) [inline app/views/api_docs/subsections/_basic_information.html.haml:158]: maybe Within a Project, the user will create resource orders and add open-access resources?

**Comment by jswk** [comment]
> link for Ordering API takes to the page with Offering API

It should be correct now.

**Review by michal-szostak — APPROVED** [review]

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**